app/code/community/Manners/Widgets/Model/Collection/Product.php

```diff
@@ -26,6 +26,7 @@ public function getFilteredByProductIds($sProductIds)
         $this->addAttributesToCollection();
 
         $this->oProductCollection->addFieldToFilter('entity_id', array('in' => $aProductIds));
+        $this->oProductCollection->setVisibility(Mage_Catalog_Model_Product_Visibility::VISIBILITY_BOTH);
```

Comment:
Should it not use something like `Mage::getSingleton('catalog/product_status')->addVisibleFilterToCollection($collection);
        Mage::getSingleton('catalog/product_visibility')->addVisibleInCatalogFilterToCollection($collection);`